### PR TITLE
AoT compatibility, CI setup

### PR DESCRIPTION
**Pull request by ryzy**
Hi,

This change introduced couple of things, here's reasoning behind them:

* `yarn` and `yarn.lock` for installing the dependencies
* `yarn build` command for compiling the project into ES2015 and with AoT support (compiling using `ngc`
* CircleCI for basic test of build

Re 1: I hope you're OK with that. If someone doesn't like the `yarn`, `npm` should still work.
Re 2 and 3: Because we don't publish the NPM package to NPM registry (ATM, see #18 and #19), the process of building and including build artefacts is semi-manual. I have added additional check in CI to make sure the compiled version of the file are in sync with source files.

Resolves: #30 
Resolves: #29

### CODE REVIEW

**Comment by ryzy** [comment]
@Pauan once you're happy with the changes and they're merged, you could link the repo with CircleCI. Nnot sure if you've used it, but the process is very easy: just go there, login with GH account, then add the project and trigger the build. Since then build will be triggered automatically for all commits and PRs.

Later on, once we have build in place, we could add build badges to README.

File: package.json [hunk 1]
```diff
@@ -1,15 +1,22 @@
...
+  "version": "1.3.0",
   "description": "Official Angular 2 plugin for amCharts V3",
+  "module": "index.js",
+  "typings": "index.d.ts",
   "repository": {
     "type": "git",
     "url": "https://github.com/amcharts/amcharts3-angular2.git"
   },
-  "peerDependencies": {
-    "@angular/core": "^2.0.0 || ^4.0.0"
```
Comment (Pauan) [inline package.json]: This `peerDependency` is still needed. Please add it back.

File: package.json [hunk 2]
```diff
@@ -1,15 +1,22 @@
 {
   "name": "amcharts3-angular2",
-  "version": "1.2.0",
+  "version": "1.3.0",
```
Comment (Pauan) [inline package.json]: Please don't bump the version. I prefer to group multiple changes into a single version.

File: README.md [hunk 3]
```diff
@@ -87,8 +87,22 @@ this.AmCharts.updateChart(this.chart, () => {
 
 You can see some examples in the `examples` directory.
 
+## Development
```
Comment (Pauan) [inline README.md]: I'm going to publish an npm package really soon, so I don't think these directions are necessary.

File: circle.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+machine:
```
Comment (Pauan) [inline circle.yml]: What's the benefit of Circle CI compared to other CIs like Travis?

File: tsconfig.json [hunk 5]
```diff
@@ -1,29 +1,26 @@
...
-        "declaration": false,
+        "declaration": true,
         "noImplicitAny": true,
-        "noImplicitUseStrict": false,
         "removeComments": true,
-        "noLib": false,
-        "preserveConstEnums": true,
         "suppressImplicitAnyIndexErrors": true,
-        "sourceMap": true,
+        "inlineSourceMap": true,
```
Comment (Pauan) [inline tsconfig.json]: Why did you change it to inline source maps?

File: package.json (hunk 2 above)
Comment (ryzy) [inline package.json]: Hey! Yeah, was guessing what's the release process. No worries, will revert this :)

File: package.json (hunk 1 above)
Comment (ryzy) [inline package.json]: Hmm, and if we have it in `devDependencie`, it's not enough?

Wasn't sure about this one.

File: README.md (hunk 3 above)
Comment (ryzy) [inline README.md]: Great to hear that, will remove then :D

File: circle.yml (hunk 4 above)
Comment (ryzy) [inline circle.yml]: Not sure really... In the past (2y ago) I've chose CircleCI for their speed, Docker support and price. And since then I'm with them. 

But.. TravisCI is OK too - would you prefer this one? Can adjust.

File: tsconfig.json (hunk 5 above)
Comment (ryzy) [inline tsconfig.json]: That's a vigilant eye! :)

No particular reason. Thought I'll be less files here and for the diff. You want me to revert this one?

File: package.json (hunk 1 above)
Comment (Pauan) [inline package.json]: We [import `@angular/core` at runtime](https://github.com/amcharts/amcharts3-angular2/blob/e708364f412cc04af45f80cf9427b22fc112e78c/index.ts#L1) so it can't be in `devDependencies`

Putting it in `peerDependencies` is better than `dependencies` because it lets the user choose which version of `@angular/core` to use, and it ensures that there is only a single version of `@angular/core`. This is the recommended way to handle plugins with npm.

File: circle.yml (hunk 4 above)
Comment (Pauan) [inline circle.yml]: I don't have a preference for CI, I just know that Travis is very popular, so I was curious about the differences.

File: tsconfig.json (hunk 5 above)
Comment (Pauan) [inline tsconfig.json]: Yes please. Having inline source maps bloats up the file in the situation when the user isn't using source maps. And some tools might not work well with inline source maps.

File: circle.yml (hunk 4 above)
Comment (ryzy) [inline circle.yml]: I'd say there's no clear winner, CircleCI is around for a while and used widely (incl. Angular project). Also see popularity in terms of search: https://trends.google.co.uk/trends/explore?q=travisci,circleci

You OK to leave it as it is?

**Comment by ryzy** [comment]
@Pauan thanks for the feedback, I made the requested amends.

**Comment by Pauan** [comment]
Looks good, thanks!
